### DISCUSSION TITLE
ELKS Install  on 8088 MB 512K  doubts

### DISCUSSION THREAD

**Discussion by lmpp1972** [jbruchon/elks#1940] [General]
Hi Everyone,
I found ELKS project today and i am trying to install it.
I got a Juko baby motherboard with 8088 CPU and 512k DRAM.
I also have a XTIDE with 64MB card formated with Dos 6.22.
I downloaded from v0.7.0 the fd360-fat.img and rawrite it to a 360k floppy , it start loading , it also found my 3Com eth503 but it stops there!

![ELKS-Hang](https://github.com/user-attachments/assets/f99e752b-a678-4c69-8b1c-1fd219a7fb1e)

Any hint will be appreciated.
Thank you :)

**Comment by ghaerr** [comment]
Hi @lmpp1972,

It seems that for some reason the ELKS NE2K network driver is hanging on your system, during the probing stage where it tries to find out more about the card. Can you edit the `/bootopts` file on the MSDOS floppy you're using to boot? If so, that would be the easiest way to disable the NE0 driver. Add or edit the `ne0=` line to `ne0=12,0` (IRQ 12, address 0), save the file and reboot.

The other way requires downloading the ELKS source and rebuilding the system with the NE2K card disabled, which is probably a lot of work for you. Of course, you could also just try pulling the network interface card for now, then edit /bootopts after ELKS boots.

Thank you!

**Reply by lmpp1972** [reply]
Hi @ghaerr,

It's not the Ethernet card that causes conflict, it's the XT-CF-Lite on I/O 300h !
If I change the I/O for XT-CF( 320/360/380)...it says master on 300h not found and i have no Disk,
but the elks linux boot from floppy(now fd360-minix.img)
I try to change eprom address but it does nothing....the card don't let me change I/O port other than 300h i try eprom write enable too....
any more ideias?
thanks
![ELKS-Floppy_no_card](https://github.com/user-attachments/assets/ca3c3f2f-c644-4337-9d09-619551b79a4e)
![XT-CF](https://github.com/user-attachments/assets/32a543a8-6821-429f-9972-6e73d1a3be56)

**Reply by lmpp1972** [reply]
Bigger photos :)
![ELKS-Floppy_no_card](https://github.com/user-attachments/assets/64edb7e2-8d02-4148-9548-f3d38edb0e9f)
![XT-CF](https://github.com/user-attachments/assets/917b3518-7e3a-4666-bf90-acb15628c0ba)

**Reply by ghaerr** [reply]
Hmmm, I'm not an expert on setting up XTIDE, but it sounds like there's something either sharing the same I/O address or possibly another setup problem. ELKS uses the BIOS for accessing the hard drive as well as querying the number of hard drives, so it is entirely dependent on the BIOS/XTIDE being setup properly.

OTOH, some time ago (I looked but wasn't able to quickly locate it), a user was trying to get XTIDE to work with ELKS and the problem ended up being that the CF card itself was the problem. Apparently there are somewhat strict requirements for compatibility and swapping the CF card out with another brand fixed the problem, finally!

**Reply by tyama501** [reply]
Hi @lmpp1972 

Nice to see it is booted!

Doesn't the ioport conflict with the ether when you have used the dos?

**Reply by tyama501** [reply]
Can the ether change io address?

**Reply by tyama501** [reply]
Hello @ghaerr 

I am not familiar with IBM ether cards,
but if we can find what the real IO port the card is using,
then we can skip the false detection of the IO 0x300, right?

**Reply by ghaerr** [reply]
@tyama501, it appears that the NE2K NIC probe routine might be doing something that hangs the system when the XTIDE is present at that address. Setting /bootopts ne0=12,0 (to a zero address) effectively skips the false detection, but @lmpp1972 has not stated whether such a method works or not. There isn't another way to turn off the NE2K probing other than to compile a kernel without NE2K support.

**Comment by lmpp1972** [comment]
Hi @ghaerr @tyama501 ,

I have changed the XT-IDE board, take the 3com Eth off(but it wasn't there on a few previous attempts and it hanged ) and changed the Compact Flash 64Mb.
And it boot  ok  detecting the Compact Flash now showing(hda,hda1) with 1000 cylinders 4 heads and 32 sectors.
The compact is imaged with hd32mbr-minix.img
I can mount it .
![boot_1](https://github.com/user-attachments/assets/0c31c5cf-b6dd-4f82-951a-cab24b61c6a0)
![boot_2](https://github.com/user-attachments/assets/c0985ca8-c388-410b-a9ee-c19be15eb443)
![boot_3](https://github.com/user-attachments/assets/ec3969a3-4ad2-42d9-a188-2e369f5a478a)

So, do i need to introduce a new line in bootopts or i just need to execute the command: makeboot /dev/hda1?
For the 3com i can use Memory jumpers from(Disable,C8000,CC000,D8000,DC000) I/O Base (300,310,330,350,250,280,2A0,2E00)
After the HD start to boot i will try the eth like 3c0=0x310,......? any sugestions because i dont know eth IRQ !
Thank you

**Reply by tyama501** [reply]
Hello @lmpp1972 Thank you.

Hello @ghaerr 
FAT filesystem sometime needs 
sys /dev/hda1
to clean partition after mkfat
and copy /linux before makeboot 
since the /linux needs to be within 8 entries, but there are no constraint for the MINIX filesystem?
Can we do makeboot to the 
existing partition?

**Reply by ghaerr** [reply]
@tyama501,

Yes, makeboot should work on an existing partition. For now, if worried about /linux not being created within the first 8 root directory entries for FAT, one can create a file using `touch /mnt/linux` that will reserve that directory entry for a later sys command that actually writes the real OS image to the same filename, or copy it beforehand as you suggest. This is all because we are quite short of extra space in the FAT boot loader.

The MINIX boot loader does not have the same constraint so just using `sys /dev/hda1` should work.

**Comment by lmpp1972** [comment]
Hi  @ghaerr, @tyama501,

Hi got it started from the Compact Flash 64Mb!

The steps were this ones
With fdisk I create a partion  started at 63 to 63504 
flagged to be boottable and write to partition table.
![bok1](https://github.com/user-attachments/assets/b72771ef-16d3-400e-b8ae-8f5a447a8505)

then create on that partition a minix file system .

![bok2](https://github.com/user-attachments/assets/ef9d3e62-6b88-4403-a543-b7494778d952)

once that done, i transfer the system from floppy to disk with sys command:

![bok3](https://github.com/user-attachments/assets/acf88b90-63ff-4845-98c4-9479ff0c3904)

I  used makeboot command to transfer the bootopts file to the disk(maybe it was already there....)

and finally i mount the HDa1 to touch the linux kernel file(I included the file on my 360k floppy)
After that it don't hang on start and loads ok

![bok4](https://github.com/user-attachments/assets/112e46e4-c417-4c43-971e-7415b608f32f)
 
   Issues :the second(B) drive i have it's a 720k 3,5" floppy and Elks think its also 360k  like (A)...if i try to mount it panics and die.
I am going to start  exploring the eth configurations.
Thank you

**Reply by ghaerr** [reply]
Wow, that is very cool @lmpp1972!! Thank you for showing the full steps you've used to get your system booting from HD.

> i have it's a 720k 3,5" floppy and Elks think its also 360k like (A)...if i try to mount it panics and die.

Can you post a screenshot of the panic? That will help me determine what is happening.

**Reply by lmpp1972** [reply]
Hi @ghaerr,
I think you don't need to worry about, i guess it was some motor issue with the drive, cause it never panic again and is mounting ok, but it detects as 360k on kernel messages right?  "/dev/fd1:40 cylinders, 2 heads, 9 sectors = 360.0 kb" 
Here it is in fat behaving well. and detected as 720k
Thank you

![720k](https://github.com/user-attachments/assets/8c1d5e14-0000-4b4f-b7bf-03da59c48af6)

**Reply by ghaerr** [reply]
> but it detects as 360k on kernel messages right? "/dev/fd1:40 cylinders, 2 heads, 9 sectors = 360.0 kb"
Here it is in fat behaving well. and detected as 720k

The floppy driver and ELKS get a bit complicated in the default configuration because ELKS uses the PC BIOS for all disk I/O, and in most cases (if the BIOS supports it) uses the BIOS to tell what the drive parameters are. So if things are working for you then I would not worry about it.

**Comment by lmpp1972** [comment]
Hi again!
For the 3com network I was using a 3com 503c so it is a 8bit card with BNC and AUI!!! and have not succeed :( 
I changed for 16bit 3com 509c with rj45 and it detected the card and assume the /bootopts entry i did 3c0=9,0x350,,0x80
Here it is detect and with the mac address etc....
![3c509](https://github.com/user-attachments/assets/e5ded8ba-58b8-4180-b0ad-da63551eac18)

I need to get /bin/*  from another card image to have all the commands,  cause they are not yet on  the flash that boots ok
![network](https://github.com/user-attachments/assets/76ebadfb-1495-4cd0-80a8-0aea4de8cf4e)

Will be support for the 3com 8bit 503c BNC connection ?
Thank you

**Reply by ghaerr** [reply]
> I need to get /bin/* from another card image to have all the commands

Looks like /dev/tcpdev is missing, see ELKS /bin/sys for the mknod command that creates it. Normally, `sys` is run to create all required device files and copy most of /etc to the new drive.

> Will be support for the 3com 8bit 503c BNC connection ?

I don't actually know. The 3com NIC driver was contributed and I'm not deeply familiar with the different chip sets and/or network interfaces it supports. Your screenshot seems to show it initializing well, but we'll need to see the results of `net start` in order to determine more.

**Reply by Mellvik** [reply]
IIRC, the 3c503 is a 8390 based adapter with a lot in common with the ne2k and wd8003, in particular the latter since it's using a shared memory architecture. It has nothing in common with the etherlink III (3c509) adapter supported by ELKS and TLVC.

**Comment by toncho11** [comment]
Juno motherboard? Is this a Pravetz 16 that you are using?

**Reply by lmpp1972** [reply]
Hi @toncho11,

I don't think these mobos are the ones you r thinking they are JuKo not Juno!
They were in local assembled(clones) cases of Portugal 
![MB2](https://github.com/user-attachments/assets/7fb1d76f-7956-4ded-89a2-6cc1b20a51d3)
![MB_JUKO](https://github.com/user-attachments/assets/4d62aeb6-b503-4b20-9c6e-9882d3d750c8)

Thanks

**Reply by toncho11** [reply]
Yes. You are correct. It is Juko for Pravetz 16 :)

**Comment by toncho11** [comment]
There is a sys command in ELKS that does installation of ELKS on hdd.

**Comment by lmpp1972** [comment]
Hi !
Just to let to know the 3com 509C 16bits worked just fine !
net start 3c0 start and the daemons also started
![1_ok](https://github.com/user-attachments/assets/0a1a4d75-9e15-4f23-a8eb-056c4559654d)

So i telnet to another machine and get connection !
![2_ok](https://github.com/user-attachments/assets/35033245-de94-4e94-b5fc-c3bfe1a157e3)
![4_ok](https://github.com/user-attachments/assets/da3fa9a8-0acd-4b53-8a7b-07e792a9ec0a)

Thank you

**Reply by ghaerr** [reply]
Thank you for your continued testing @lmpp1972! Its nice to continue to see ELKS coming up and running properly on more systems. I'm glad to see that the 3Com NIC is working fine :)

**Comment by tyama501** [comment]
Hi @lmpp1972 ,

Does your system and the monitor support CGA 640x200, 2 colors, interlace?
I have made a nano-X driver recently for the graphic mode.
https://github.com/ghaerr/elks/discussions/1939#discussioncomment-10235745

Here is the nxtetris I compiled with the driver. (I renamed it as nxtetric for now)
[nxtetric.zip](https://github.com/user-attachments/files/16512187/nxtetric.zip)

I have only tested at emulators.
It is nice if you can try it with the real PC-XT :)

The key is
n : new game
p : pause
s : rotate
j : left
k : right
space : drop
q : quit

**Comment by lmpp1972** [comment]
Hi !
Can someone explain how to change from tty console to tty1 ?Ctrl+Alt+Fx does nothing to me....
I have change on /etc/inittab
initdefault to 5
but I have to comment #t2: line and # t3 or i get a boot error /bin/init: Can't open /dev/tt2 /dev/tty3 (errno 19)
 I would like to know how to create users too ......I see that are on passwd entries for at least one User1, i guess!
Thank you
![feh_002091_000001_inittab](https://github.com/user-attachments/assets/95f7e319-44a8-404e-9948-c7bac4eb5f9f)

**Reply by ghaerr** [reply]
> Ctrl+Alt+Fx does nothing to me...

Yes, you'll need to sort out why changing to changing to init level 5 isn't working first. BTW, you can run `init 5` instead of editing /etc/inittab if you'd like.

> but I have to comment #t2: line and # t3 or i get a boot error /bin/init: Can't open /dev/tt2 /dev/tty3 (errno 19)

Yes, remove the comment on t2 and t3. Errno 19 means device does not exist: it seems for whatever reason that /dev/tty1 and /dev/tty2 don't exist on your system. You can use ls -l /dev/tty* to check that.

To create them, use `mknod /dev/tty2 c 4 1` and `mknod /dev/tty3 c 4 2`.

> I would like to know how to create users too

We do it the old fashioned way, by editing /etc/passwd directly. Copy the line user2 and create another named user, and increment the numbers from 102 to 103. After saving the file, you should be able to login as that user and then run `passwd` in order to set a password for that user. That will also be stored in /etc/passwd in encrypted form.

**Comment by lmpp1972** [comment]
Hi! @ghaerr 

Have deleted and created new entries with mknod for tty2 and 3, the process runs normally but the error appears at the boot time if uncommitted .
Still don't know why Ctrl+Alt+F1,2,3 don't work with init 5 enabled. 
The users and passwd were created with success, i also can telnet the machine with my user :)
Still playing....have upgraded from 512k to 640k with Apro ISA PC/XT 512k Ram Expansion bought on ebay.
As far as I understand since i have 512k on board i only can use + 128k from the card....
![2024-08-20-205720_484x1109_scrot](https://github.com/user-attachments/assets/2026a335-f939-4fc1-8588-fbedfffea7d8)
![card_front](https://github.com/user-attachments/assets/0b678ba5-69ee-45fe-9cf2-6e31c7e692c4)
![card_back](https://github.com/user-attachments/assets/315d8751-2301-4d93-988e-9c3d575f5f81)
Thank you

**Reply by ghaerr** [reply]
> Have deleted and created new entries with mknod for tty2 and 3, the process runs normally but the error appears at the boot time if uncommitted .

I don't understand what you mean "uncommitted"? Do you mean that it works after you do the mknod but then after boot it fails again? You'll have to make sure to "sync" before rebooting to sync buffers to disk, but that may not be what you're talking about.

> Still don't know why Ctrl+Alt+F1,2,3 don't work with init 5 enabled.

This is likely due to /bin/getty not being able to open /dev/tty2 and 3. You can run `ps` to see if `getty` shows up - if they do, then things are working, if not, then likely getty can't even open them, and exits. Run `ls -l /dev/tty*` after boot to see what that looks like.

That is a very cool screenshot! And I really like your `elksfetch` shell script - care to post it? :)

> As far as I understand since i have 512k on board i only can use + 128k from the card

I haven't studied your switch settings but if you have the external RAM configured just above 512K for 128K, `meminfo` should show a total of just under 640k total instead of the 495k shown now.

After you get the RAM working, ELKS can also use an additional 64K of UMB memory at 0xC000. That's configured in /bootopts where there's an example of it.

**Reply by tyama501** [reply]
Hi @lmpp1972 

Wow, the elksfetch really cool!

Hi @ghaerr 
We can use more than 64KB on UMB if available right?
I think I have been using more than 64KB UMB on pc-98 (by using hardware ems as umb with special free software).
The disk buffers  and some other thing are moved to umb.

**Comment by tyama501** [comment]
I found the picture before using umb and after using umb.

Before
![IMG_20240615_151654](https://github.com/user-attachments/assets/a257cc01-795e-4980-acdc-8858f911645e)

After
![IMG_20240615_150831](https://github.com/user-attachments/assets/6e3f9d83-ca05-474d-80c9-42a85a01cc8b)

**Comment by lmpp1972** [comment]
Hi @ghaerr 

Here is the script  to anyone who wants to improve....

![2024-08-20-224459_484x875_scrot](https://github.com/user-attachments/assets/36a4f2e3-4837-4dd9-a98f-ca4a55165ffe)

elksfetch 
#Distro logos and ASCII outputs

#GREEN='\033[0;32m'
#NC='\033[0m' # No Color

#tput setaf 1;

echo "${c1}      ___  _    _ __ ___     "${c1}"$USER""@$HOSTNAME"
echo "${c1}     | __>| |  | / // __>    "${c1}"OS: ""ELKS Linux"
echo "${c1}     | _> | |_ |  \ \__ \    "${c1}"Kernel: " "$(uname -m -r)"
echo "${c1}     |___>|___||_\_\<___/    "${c1}"Uptime: " "$(uptime)"
echo "${c1}                             "${c1}"Shell:  " "($SHELL)"
echo "${c1}                             "${c1}"Processes:""$(ps axu | wc -l)"

#tput setaf sgr 2;
#echo -e "\033[0m"

Well tabs don't look good here!
But the url is this one with Font name: Dr Pepper !
https://patorjk.com/software/taag/#p=testall&h=1&v=3&f=Small&t=ELKS
Just paste line by line and ajust inside the editor with tabs

**Reply by toncho11** [reply]
Post it as git hub code and it should keep the tabs.

**Reply by lmpp1972** [reply]
@toncho11 

Have to learn a lot in using github :)  i wish i knew to do that .....
Thanks

**Reply by ghaerr** [reply]
> Have to learn a lot in using github :) i wish i knew to do that .....

If you put a file (shell script, C code, whatever) in a .zip file, you can then drag that into one of these comment boxes and it will upload the file.

> https://patorjk.com/software/taag/#p=testall&h=1&v=3&f=Small&t=ELKS

Wow, that's quite a cool ASCII text generator! :)

> We can use more than 64KB on UMB if available right?

Yes, as long as it doesn't overlap video memory or the BIOS, of course. ELKS doesn't support special/DOS extenders etc, the memory just needs to be real-mode addressable wherever the /bootopts umb= statement says it is.

**Comment by lmpp1972** [comment]
Hi @ghaerr 
Even after the sync command and reboot no Ctrl+Alt+Fx doesn't work .... 
getting the error:
/bin/init:
/bin/init: Can't open /dev/tty3 (errorno ECan't open19)
/dev/tty2 (errno L19)
KS 0.7.0

login:

ps :where can write net start 3c0 to start on boot ...have tried bootopts but it not started
Can we create a ~/.shrc  file ?
Thanks

**Comment by ghaerr** [comment]
> Even after the sync command and reboot no Ctrl+Alt+Fx doesn't work ...

Ah, I see now. In your very first screenshot, ELKS says "Direct console, scan kbd 80x25, emulating ANSI (1 virtual console)" - which means the kernel console driver set you up for only one virtual console. That's the problem.

Looking at console-direct.c source:
```
    VideoSeg = 0xb800;
    if (peekb(0x49, 0x40) == 7) {
        VideoSeg = 0xB000;
        NumConsoles = 1;
        isMDA = 1;
    }
```
It appears that your console is MDA!? The code above checks for BIOS mode 7, which is 80x25 monochrome text at video address 0xB000, instead of BIOS mode 3, which is 80x25 color text at 0xB800. Can you change your BIOS setting to startup in Mode 3? ELKS doesn't try to set a video mode, and starts using whatever video mode the BIOS has as default on boot.

> Can we create a ~/.shrc file ?

You can edit /etc/profile for all users. I can't recall but I think the shell also supports ~/.profile (MINIX filesystems only).

> where can write net start 3c0 to start on boot ...have tried bootopts but it not started

See the sample /bootopts: put `net=3c0` on a blank line in /bootopts.

Also take a look at /etc/rc.sys to see how that works, you can put stuff in there too.

**Comment by lmpp1972** [comment]
Hi @ghaerr @tyama501 

Just a little update on the MDA to CGA card. 
To use it I had to change monitor also... Now is a Ibm 5153.
So I bought one of those China made 8bit  CGA cheap card  and it works :) I now have 3 virtual consoles also CTRL+ALT+F'123'
I can play nxtetric and is very well playable ...is fast ! 

Here the images! 
Thank you all

![china-cga](https://github.com/user-attachments/assets/30be1d7f-cf6d-44eb-99ff-c76632103486)
![bios_640](https://github.com/user-attachments/assets/fe0bf2af-383d-459b-a8d8-cf7700c17823)
![Elks-640k](https://github.com/user-attachments/assets/a7856c26-68ef-42f6-89b9-f809d9e3f344)
![nxtetric](https://github.com/user-attachments/assets/dbca235b-6877-4f8b-bbda-e6431e86dc1a)
![Virtual-console](https://github.com/user-attachments/assets/fa8affdb-1847-4a23-bfd4-f79524f8480b)

**Reply by toncho11** [reply]
I have a color IBM 5153.
So I need both this Chinese card and I need to disable CONFIG_HW_VGA while building ELKS which will force nxtetris to compile with CGA support instead of VGA?

**Reply by lmpp1972** [reply]
I did nothing to  get it run, is  @tyama501 original binary .
Only thing i change was the CGA card, 
On MDA it not run or show the mouse pointer.

**Reply by tyama501** [reply]
Hi @lmpp1972 

Nice! Thank you!

**Reply by ghaerr** [reply]
@lmpp1972, this all looks great! :)

I think what @toncho11 is wanting to know is which version of nxtetris is running on your CGA? Are you using the nxtetris that came on the ELKS v0.7.0 distribution, or are you using the specially-created binary of nxtetris that @tyama501 just built a few weeks ago? 

My understanding is that in order for the new nxtetris to run on CGA, it has to be built specially without CONFIG_HW_VGA=y when one builds all the ELKS applications from the current repo, but I'm not fully sure myself.

**Reply by toncho11** [reply]
Bravo @tyama501 !

**Reply by tyama501** [reply]
Thanks @toncho11 

Hi @lmpp1972 
BTW
You can hide the mouse pointer by export MOUSE_PORT=none
if you don't use the mouse.

Thank you.

**Comment by lmpp1972** [comment]
@ghaerr  @toncho11 
Here is the elkfetch.zip 
[elksfetch.zip](https://github.com/user-attachments/files/16719032/elksfetch.zip)

Needs improvements .....
Thank you

**Comment by lmpp1972** [comment]
Hi @ghaerr !
I have now a WD8003EBT 1987 8bits AUI/BNC to experiment on Elks 0.7.0
And didnt found a good line to enter on bootopts  and put it to work! 
 
So  i send you a pdf file and on page 262-263 we can see information about this board all  jumpers sets , I/O address and RAM base addresses .
Anyway i still couldn't  found a way to get it work.... any suggestion is apreciated.
I think i got  this config on the card:
W1
1-2 Open
7-8 Closed
9-10 Closed
W2
1-2 Close (IRQ 2)
W3
All Closed
W6
1-2 Closed
3-4 Closed
5-6 Closed
W7
3-4 Closed
W8
3-4 Closed
W9
1-2 Closed
W10
1-2 Closed
W11
2-3 Closed
Thank you

Card image : 
![20240830_212336](https://github.com/user-attachments/assets/a687e7d8-3ebd-470d-9da2-c800c41db9e5)
PDF zip file: 
[pc_engineers_vol3_misc.pdf.zip](https://github.com/user-attachments/files/16821903/pc_engineers_vol3_misc.pdf.zip)

**Reply by ghaerr** [reply]
Hello @lmpp1972,

> And didnt found a good line to enter on bootopts and put it to work!

Are you using the uncommented wd0= /bootopts line, but filled in with your IRQ and NIC ram base address?

I'm not an expert on network settings like @Mellvik is. @Mellvik, do you have any experience you can offer on this?

**Reply by Mellvik** [reply]
I have a working card almost like this one, using the AUI connector with an AUI to TP adapter. This is the reference I've been using
http://www.stack.nl/~marcolz/network/wd80x3.html#WD8003EP

I can send you the jumper settings I am using (picture) and the bootopts line tomorrow, I'm not sure the closeup I have here covers everything.
![IMG_9871](https://github.com/user-attachments/assets/8ad287d5-6eca-42ba-8d27-ba1e938a199d)

**Comment by lmpp1972** [comment]
Hi @ghaerr  @Mellvik 
Sorry the delay  i have been without my BSD system( done a major update and got some breaks...working now)

Yes, I tried with the default line in bootopts , just changing the IRQ to 2 cause mine I set it to 2
not working !

eth: wd0 at 0x300, irq 2, ram 0xcc00 not found

I guess i need to pass I/O 2C0 or 2E0  and for ram address C0000, C4000 or C8000

I need to use BNC connection , I am using a 10mbit router with 1x BNC.

Maybe @Mellvik could show the line with the values for ram ?

Thank you very much

**Comment by Mellvik** [comment]
@lmpp1972 - 
I have this card running in a IBM 5155 with the following `bootopts` line:
```
wd0=2,0x280,0xCC00,0x80
```
Also, as it turns out all available jumpers on my card are visible on the picture I sent yesterday. It seems our hard has BNC activated since all jumpers are closed on W3.
From the docs in the link I sent you, it seems the following  settings should work for your card - with the config line above:

- W3 is ok, all closed
- W5: closed
- W4 if it exists: Open
- W1 for IO port 280: close 3-4,5-6, 9-10 (these are also the factory defaults)
- W2 for IRQ 2: close 11-12
- W6 (disable ROM), close 7-8, 9-10
- W9: 3-4
- W10: 3-4
- RAM - it looks like your card has 32K ram. This is good but I don't think the driver actually takes advantage of more than 8k. Anyway, try these settings:
- W7: 1-2
- W8: 1-2
- W11: 2-3

There is no jumper for the shared memory address, it will have to be sett using the EZSETUP utility running MSDOS. However, only a few addresses are possible, you can try them by changing the setting in the `bootopts` line: C000, C800, D000, D800.

Hope this helps.

-M

**Comment by lmpp1972** [comment]
Hi @Mellvik 

Thank you for your quick response and for the tips!

I have done all the jumpers has you mentioned and the card is found but hangs!

Then i open the W5 and with it open it not hangs, but give errors about damage packets....it was at lunch time and i did not have much time to test more.....i think i have to play with w7 and w8.
 Anyway at least it  is alive...thank you for any advice

here W5 closed it hangs
![20240902_131634](https://github.com/user-attachments/assets/f2c75308-649f-4887-9df9-ea05917db4c0)

W5 open and commented #net=wd0 at bootopts .....it goes to console 
![20240902_134138](https://github.com/user-attachments/assets/7cd5192e-7f60-4fe1-9fdb-bc7df6b38ccd)

error message ....
![20240902_131634](https://github.com/user-attachments/assets/67f73c54-57a9-47e4-95b5-919cd7a4bc58)

**Reply by Mellvik** [reply]
Hi @lmpp1972 - this is looking good. 
I don't quite understand what you're saying though. Which of the screenshots have the bootopts configuration line activated? You need to use the bootopts configuration in order to find out which shared memory address to use, unless you configure it using EZSETUP as I described before.
And I don't see any error messages on any of the screens. Could you elaborate on that?

Removing the W5 jumper is likely the reason you get errors about damaged packets. That W5 should cause the driver to hang seems unlikely.

Also, if the card seems be found during boot when you have disabled the wd0 line in bootopts, it means that you've set the card address to 0x240 - the default address in ELKS, which is fine. But then you need to change the setting in bootopts to reflect that:
```
wd0=2,0x240,0xC800,0x80
```
Notice also that I changed the address to 0xc800, the 0xcc00 address will not work with a 39kb ram card.

Let me know how it works out.

**Comment by lmpp1972** [comment]
Hi @Mellvik I posted twice the same screenshot ...my bad sorry!

Later at night I am going to make more tests, thank you very much.
![20240902_135101](https://github.com/user-attachments/assets/98331247-767d-4a1e-a1c9-6913b10c7abe)

**Reply by Mellvik** [reply]
Thank you @lmpp1972,

seeing this I no longer think this is related to the W5 jumper. This looks more like  the buffer address being wrong. Everything else looks good. If this picture is a boot without the bootopts line active, my guess is that using the line I sent you last will work.

Let me know.

-M

**Comment by lmpp1972** [comment]
Thank you very much @Mellvik  and @ghaerr for your help once again , i couldn't go so far without you !

So  I changed the line on bootopts to wd0=2,0x280,0xD800,0x80 and it works ok

"W6 (disable ROM), close 7-8, 9-10"
and one more jumper on 5-6 (close) I think for D8000(16k,32k ROms), if i add also another jumper on 3-4 (close) it gives error Bad Cheksum.....

booting with #net=wd0 or net=wd0 not hang anymore.

![20240902_1](https://github.com/user-attachments/assets/a80e6c18-c688-41a3-96aa-535539b73d67)

![20240902_2](https://github.com/user-attachments/assets/98457e3e-8d30-48fc-970b-48b46b35e897)

Telnet connections on my router and a BBS!
![20240902_3](https://github.com/user-attachments/assets/7a5b380d-da7c-4766-8b37-f58632f8199e)
![20240902_4](https://github.com/user-attachments/assets/d6ec467b-72a9-4723-8b46-e78d0846350c)

FTP to my I486 redhat commercial 1.1 mother's Day
![20240902_5](https://github.com/user-attachments/assets/20ddaf28-e476-4097-89b3-0e6b162c2b88)

I think it works perfect on BNC cable and switch!
![20240903_6](https://github.com/user-attachments/assets/acae031e-d30d-4b75-8b93-9bd8df511338)

**Reply by ghaerr** [reply]
Wow @lmpp1972, you've done a great job getting all these ELKS features working on your system. I love the BBS screenshot :)

Thanks for your help on this @Mellvik, it is greatly appreciated!

**Comment by Mellvik** [comment]
@lmpp1972 - congratulations!!

Now, here's something to try: Set the last parameter in the configuration line (this is the 'flag' field) to 0x83 instead of 0x80. This is supposed to tell the driver to take advantage of 32k buffer space on the NIC, but it has never been tested on real hardware.

If you get errors, change to 0x82 (16k). If you still get errors, go back to what you have now.

Please report back what you experience. That way you contribute to the testing of the driver and thus the development of the system.

-M

**Comment by lmpp1972** [comment]
Hi @ghaerr  and @Mellvik !

So i tried 0x83 and i got errors and also with 0x82

I mean errors using ftp or telnet 

0x83 FTP  tcp: BAD CHECKSUM (0x76f2 len 907)
![Bad1](https://github.com/user-attachments/assets/b25c1c64-e38e-453b-958e-bc8673c93e21)

0x83 errors using telnet >
![Bad2](https://github.com/user-attachments/assets/af402f17-af61-49e3-804b-3c80cd7bf50a)

0x82 ftp
![Bad3](https://github.com/user-attachments/assets/06d27f82-1522-4a5f-a6dc-6af02f8f9721)

0x82 telnet ...you see the error in green near the eyes of the slime creature:) BAD CHECKSUM (0x3825) len 1480
![Bad4](https://github.com/user-attachments/assets/05c088e3-dff1-4d71-bc48-f298903e631c)

So what you think .is better  to stick with 0x80 or still  is something  misconfigured?
Thank you

**Reply by Mellvik** [reply]
Thank you @lmpp1972 for testing.

Please revert to 0x80 as you suggest. I'll take a close look at the driver and the switch settings again. If more testing is useful, I'll let you know.

-M

**Comment by tyama501** [comment]
Hello @lmpp1972 

I have added CGA support of the BASIC in #2003 .

I attached the built BASIC here since you may be interested in this.
I renamed it to basicga for now.
[basicga.zip](https://github.com/user-attachments/files/16997353/basicga.zip)

Here is some examples to draw. (TEST1.BAS, TEST2.BAS, TESTCGA.BAS)
[TESTCGA_BAS.zip](https://github.com/user-attachments/files/16997387/TESTCGA_BAS.zip)
These can be loaded like
load "test1"
on the basic.
Hitting Enter will exit the example program.

CTRL-D will get back to the text mode and exit the basic.
(CTRL-C stop the running program,
but currently it does not get back to text mode so the display crashes if it is still in the graphic mode.)

Also, the snake game will be uploaded with this PR #2007 .
(I haven't fixed the 7 character name issue on the PR yet.)
 
Further information for the basic can be found here.
https://github.com/ghaerr/elks/blob/master/elkscmd/basic/README.md

Thank you.

**Reply by toncho11** [reply]
@tyama501 Can you please add more info on the subject of Basic and CGA here: https://github.com/ghaerr/elks/wiki/Coding-games-for-ELKS. There is a section about Basic.

**Reply by tyama501** [reply]
Hi @toncho11 

I have written notes and samples in 1. Using nano-X and 3. Using Basic.
Please check it.

Thank you.

**Reply by tyama501** [reply]
I also added links to the basic README and snakecga.bas.

**Comment by lmpp1972** [comment]
Hi everyone!
Have downloaded those files and have no success to run them! 
Maybe my bad or my setup....but i tried :)

Even locally it gives some errors.

![2024-09-28-175341_636x1417_scrot](https://github.com/user-attachments/assets/e3dcbebb-bd96-440b-b6db-8966b1a037ce)
Elks on startup sees the 640k PC/XT class machine but on meminfo it doesn't show!!!

I have another Juko 8088 motherboard with 1Mb DRAM onboard I'll switch soon!

**Reply by toncho11** [reply]
First you can not run graphics stuff remotely.  This is direct VGA access. You can execute it remotely ... but it will still appear on the monitor attached to ibmpc. Try on the machine without telnet. But I am not sure if this is your only problem.  
Also for ELKS Doom you need more than 500kb free memory.  Check this: https://github.com/ghaerr/elks/wiki/Increase-available-memory

**Reply by ghaerr** [reply]
You have two separate issues here. The first is that TESTCGA.BAS probably has to be changed to lower case testcga.bas, or if that is already the case, then run `basic testcga` instead (you will need to specify the full path to the testcga.bas file). I am guessing you are running the MINIX filesystem, and filenames are case significant.

The elksdoom.os2 issue is that the kernel needs to be recompiled for OS/2 Executable support (CONFIG_EXEC_OS2=y) in order to run .os2 files, and Doom has to be compiled in OS/2 format since it is so large. Version 0.7.0 did not have OS/2 executable support, but was added a couple months before v0.8.0 release. By default, version 0.8.0 has OS/2 support ON by default, which should work, although you might want to read the [Wiki article](https://github.com/ghaerr/elks/wiki/Increase-available-memory) about setting /bootopts for max memory availability.

**Reply by ghaerr** [reply]
> Also for ELKS Doom you need more than 500kb free memory.

Yes, this is true. The current error message is "Syntax error", for which the solution is as I described above. If however, you get "Not enough memory", then follow @toncho11's instructions.

**Reply by tyama501** [reply]
Hi @lmpp1972 
Thank you for trying.

The basic I have attached above is basicga.
Did you renamed it to basic?

**Reply by toncho11** [reply]
So you need a different basic and you also need code changes? I mean there is snake and snakecga, right?
There should probably be only one basic executable that is compiled for cga for non PC-98 images and one that is compiled with compatibility for PC-98.
Also the change in the basic code is because in CGA you do not have mixed text/graphical mode?
Is it possible to have only one source file for both modes?

**Reply by ghaerr** [reply]
> There should probably be only one basic executable that is compiled for cga for non PC-98 images and one that is compiled with compatibility for PC-98.

Definitely agreed. With @tyama501's changes for CGA, the default v0.8.0 `basic` supports CGA for IBM PC, while the PC-98 basic runs PC-98 graphics. I'm not sure about the compatibility of the various BASIC .bas programs, but think that there ideally should not be different .bas programs for each system, other than changes in supported resolutions (e.g. width and height). We might need a mechanism to allow the .bas program to determine if it is running on PC or PC-98?
